packages/siopv2-oid4vp-op-auth/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -344,32 +357,62 @@ export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
 
     const pex = new PEX()
     const verifiableCredentialsWithDefinition: Array<VerifiableCredentialsWithDefinition> = []
+    const dcqlCredentialsWithCredentials: Map<DcqlCredentialRepresentation, UniqueDigitalCredential> = new Map()
+
+    if (authorizationRequestData.presentationDefinitions !== undefined && authorizationRequestData.presentationDefinitions !== null) {
+      authorizationRequestData.presentationDefinitions?.forEach((presentationDefinition) => {
+        const { areRequiredCredentialsPresent, verifiableCredential: verifiableCredentials } = pex.selectFrom(
+          presentationDefinition.definition,
+          selectedCredentials.map((udc) => udc.originalVerifiableCredential!),
+        )
+
+        if (areRequiredCredentialsPresent !== Status.ERROR && verifiableCredentials) {
+          const uniqueDigitalCredentials: UniqueDigitalCredential[] = verifiableCredentials.map((vc) => {
+            // @ts-ignore FIXME Funke
+            const hash = computeEntryHash(vc)
+            const udc = selectedCredentials.find((udc) => udc.hash == hash)
+
+            if (!udc) {
+              throw Error('UniqueDigitalCredential could not be found')
```

Comment:
throw error vs Promise.reject(Error) below (which is prefered)

Reply:
Refactored to use Promise.reject(Error)